### PR TITLE
refactor(dao) : LetterSoundContributionEvent #1677

### PR DESCRIPTION
**Pull request by SnehaHS65**
refs : #1677 
Refactor LetterSoundCorrespondenceContributionEvent -> LetterSoundContributionEvent

### CODE REVIEW

**Comment by SnehaHS65** [comment]
@jo-elimu , Please review my PR to rename only LetterSoundCorrespondenceContributionEvent -> LetterSoundContributionEvent.

File: src/main/java/ai/elimu/web/content/letter_sound/LetterSoundEditController.java [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import java.util.List;
 import javax.validation.Valid;
 
+import ai.elimu.model.contributor.LetterSoundContributionEvent;
```
Comment (vuriaval) [inline src/main/java/ai/elimu/web/content/letter_sound/LetterSoundEditController.java:6]: Why is this import duplicated?

File: src/main/java/ai/elimu/web/content/letter_sound/LetterSoundCreateController.java [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import java.util.List;
 import javax.validation.Valid;
 
+import ai.elimu.model.contributor.LetterSoundContributionEvent;
```
Comment (vuriaval) [inline src/main/java/ai/elimu/web/content/letter_sound/LetterSoundCreateController.java:6]: Another duplicate here

File: src/main/java/ai/elimu/util/db/DbContentImportHelper.java [hunk 3]
```diff
@@ -29,12 +29,8 @@
 import ai.elimu.model.content.StoryBookParagraph;
 import ai.elimu.model.content.Word;
 import ai.elimu.model.content.multimedia.Image;
-import ai.elimu.model.contributor.Contributor;
-import ai.elimu.model.contributor.LetterContributionEvent;
-import ai.elimu.model.contributor.LetterSoundCorrespondenceContributionEvent;
-import ai.elimu.model.contributor.NumberContributionEvent;
-import ai.elimu.model.contributor.StoryBookContributionEvent;
-import ai.elimu.model.contributor.WordContributionEvent;
+import ai.elimu.model.contributor.*;
```
Comment (vuriaval) [inline src/main/java/ai/elimu/util/db/DbContentImportHelper.java:32]: Wildcard imports are a bad practice

File: src/main/java/ai/elimu/web/content/peer_review/LetterSoundPeerReviewEventCreateController.java [hunk 4]
```diff
@@ -6,8 +6,9 @@
 import ai.elimu.model.content.Letter;
 import ai.elimu.model.content.LetterSoundCorrespondence;
 import ai.elimu.model.contributor.Contributor;
+import ai.elimu.model.contributor.LetterSoundContributionEvent;
```
Comment (vuriaval) [inline src/main/java/ai/elimu/web/content/peer_review/LetterSoundPeerReviewEventCreateController.java:9]: Another duplicate here

File: src/main/resources/db/migration/2004011.sql [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# 2.4.11
+
+# "LetterSoundCorrespondenceContributionEvent" → "LetterSoundContributionEvent"
+DROP TABLE `LetterSoundContributionEvent`;
```
Comment (vuriaval) [inline src/main/resources/db/migration/2004011.sql:4]: This should be made in a safer manner, like `DROP TABLE IF EXISTS`

File: src/main/java/ai/elimu/dao/jpa/LetterSoundContributionEventDaoJpa.java [hunk 6]
```diff
@@ -1,59 +1,61 @@
 package ai.elimu.dao.jpa;
 
-import ai.elimu.model.contributor.LetterSoundCorrespondenceContributionEvent;
+import ai.elimu.model.contributor.ContributionEvent;
```
Comment (vuriaval) [inline src/main/java/ai/elimu/dao/jpa/LetterSoundContributionEventDaoJpa.java:3]: Where is this new import used?

**Review by vuriaval — CHANGES_REQUESTED** [review]

**Review by vuriaval — CHANGES_REQUESTED** [review]
Please do not resolve comments that have not been addressed

**Comment by SnehaHS65** [comment]
> Please do not resolve comments that have not been addressed

I have addressed your comments in the latest commit. Can you please check

File: src/main/java/ai/elimu/util/db/DbContentImportHelper.java (hunk 3 above)
Comment (vuriaval) [reply src/main/java/ai/elimu/util/db/DbContentImportHelper.java:32]: This has not been addressed

**Review by vuriaval — CHANGES_REQUESTED** [review]

File: src/main/java/ai/elimu/util/db/DbContentImportHelper.java (hunk 3 above)
Comment (SnehaHS65) [reply src/main/java/ai/elimu/util/db/DbContentImportHelper.java:32]: Even if I add it separately, it is taking as *

Comment (vuriaval) [reply src/main/java/ai/elimu/util/db/DbContentImportHelper.java]: Please review your IDE configuration, it probably collapsed them together by default

Comment (SnehaHS65) [reply src/main/java/ai/elimu/util/db/DbContentImportHelper.java]: resolved it

**Review by vuriaval — APPROVED** [review]

**Comment by jo-elimu** [comment]
@SnehaHS65 Your migration script ran successfully in production:

```bash
[HIN] 23:46:03.987 [main] INFO  ai.elimu.util.db.DbMigrationHelper - performDatabaseMigration
[HIN] 23:46:03.987 [main] INFO  ai.elimu.util.db.DbMigrationHelper - pomVersion: 2.4.11
[HIN] 23:46:03.988 [main] INFO  ai.elimu.util.db.DbMigrationHelper - pomVersionAsInteger: 2004011
[HIN] 23:46:04.373 [main] INFO  ai.elimu.util.db.DbMigrationHelper - dbMigrations.size(): 42
[HIN] 23:46:04.374 [main] INFO  ai.elimu.util.db.DbMigrationHelper - versionOfMostRecentMigration: 2004010
[HIN] 23:46:04.374 [main] INFO  ai.elimu.util.db.DbMigrationHelper - Looking up pending DB migrations after version 2004010
[HIN] 23:46:04.374 [main] INFO  ai.elimu.util.db.DbMigrationHelper - Looking up file "db/migration/2004011.sql"...
[HIN] 23:46:04.374 [main] INFO  ai.elimu.util.db.DbMigrationHelper - Migration script found for version 2004011
[HIN] 23:46:04.379 [main] INFO  ai.elimu.util.db.DbMigrationHelper - Executing sql: DROP TABLE IF EXISTS `LetterSoundContributionEvent`;
[HIN] 23:46:04.415 [main] INFO  ai.elimu.util.db.DbMigrationHelper - Executing sql: ALTER TABLE `LetterSoundCorrespondenceContributionEvent` RENAME `LetterSoundContributionEvent`;
[HIN] 23:46:04.428 [main] INFO  ai.elimu.util.db.DbMigrationHelper - Updating current version (2004011)
[HIN] 23:46:04.499 [main] INFO  ai.elimu.util.db.DbMigrationHelper - Database migration complete!
```

**Pull request merged**